libzmq/src/error.rs

```diff
@@ -178,24 +167,19 @@ impl<T> From<Error<T>> for io::Error {
 
 /// Used to give context to an `Error`.
 ///
-/// # Note
-/// This error type is non-exhaustive and could have additional variants
-/// added in future. Therefore, when matching against variants of
-/// non-exhaustive enums, an extra wildcard arm must be added to account
-/// for any future variants.
-///
 /// [`Error`]: enum.Error.html
-#[derive(Debug, Copy, Clone, PartialEq, Eq, Fail, Hash)]
+#[derive(Debug, Error, Copy, Clone, PartialEq, Eq, Hash)]
+#[non_exhaustive]
```

Comment:
`#[non_exhaustive]` bumps MSRV to 1.40. Is this fine?

Reply:
Sure.